org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/DetectVMInstallationsJob.java

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.jdt.internal.launching;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jdt.launching.IVMInstall;
+import org.eclipse.jdt.launching.IVMInstallType;
+import org.eclipse.jdt.launching.JavaRuntime;
+import org.eclipse.jdt.launching.VMStandin;
+import org.eclipse.osgi.util.NLS;
+
+class DetectVMInstallationsJob extends Job {
+
+	private static final Object FAMILY = "Family " + DetectVMInstallationsJob.class.getName(); //$NON-NLS-1$
+	
+	DetectVMInstallationsJob() {
+		super(LaunchingMessages.lookupInstalledJVMs);
+	}
+
+	@Override
+	protected IStatus run(IProgressMonitor monitor) {
+		Collection<File> rootDirectories = new HashSet<>();
+		if (!Platform.OS_WIN32.equals(Platform.getOS())) {
```

Comment:
Please check `org.eclipse.jdt.internal.debug.ui.jres.InstalledJREsBlock.search()` for special Mac code. I have no idea about Macs, but may be you would be inspired by that.

Reply:
Check out: https://stackoverflow.com/questions/15826202/where-is-java-installed-on-mac-os-x
for description of logic.  It appears that on Mac, JRE installs are not put in the JavaVirtualMachines folder specified later in this file and you need to use /usr/libexec/java_home.   That said, why not simply change the code on MacOS to call: MacInstalledJREs.getInstalledJREs() which is static, uses the proper logic, and does all the work for you to create an array of VMStandin?

Reply:
I really cannot properly do the job for mac. There are too many things I don't know and if I do it wrong, it causes more bugs that it fixes issues.
For example: does MacInstalledJREs add standins for already known VMs? Am I supposed to filter the lists? Are the installations referencing the Java installation home or some subfolder that would still be valid? Do I have to verify the name is unique or is it already done...
There are too many things I'm not comfortable in implementing just by guess; I would rather see this patch merged as it, or even disabled on Mac, and let someone with good knowledge of mac integration and ability to test fill the holes.

Reply:
Open a new issue for Detect JVM installs at startup on Mac.  I'll look at it for M2.

Reply:
Done with https://github.com/redhat-developer/vscode-java/pull/3152